samples/ui-kit-place-search-nearby/index.ts

```diff
@@ -6,132 +6,132 @@
-const placeList = document.querySelector("gmp-place-list") as any;
-const typeSelect = document.querySelector(".type-select") as any;
-const placeDetails = document.querySelector("gmp-place-details") as any;
-const placeDetailsRequest = document.querySelector('gmp-place-details-place-request') as any;
+// Query selectors for various elements in the HTML file.
+const map = document.querySelector('gmp-map') as google.maps.MapElement;
+const placeSearch = document.querySelector('gmp-place-search') as any;
+const placeSearchQuery = document.querySelector(
+    'gmp-place-nearby-search-request'
+) as any;
+const placeDetails = document.querySelector('gmp-place-details-compact') as any;
+const placeRequest = document.querySelector(
+    'gmp-place-details-place-request'
+) as any;
+const typeSelect = document.querySelector('.type-select') as HTMLSelectElement;
 /* [END maps_ui_kit_place_search_nearby_query_selectors] */
-let markers = {};
-let infoWindow;
-
-async function initMap(): Promise<void>  {
-    await google.maps.importLibrary('places');
-    const { LatLngBounds } = await google.maps.importLibrary('core') as google.maps.CoreLibrary;
-    const { InfoWindow } = await google.maps.importLibrary('maps') as google.maps.MapsLibrary;
-    const { spherical } = await google.maps.importLibrary('geometry') as google.maps.GeometryLibrary;
-
-    infoWindow = new InfoWindow;
-    let marker;
-
-    function getContainingCircle(bounds) {
-        const diameter = spherical.computeDistanceBetween(
-            bounds.getNorthEast(),
-            bounds.getSouthWest()
-        );
-        const calculatedRadius = diameter / 2;
-        const cappedRadius = Math.min(calculatedRadius, 50000); // Radius cannot be more than 50000.
-        return { center: bounds.getCenter(), radius: cappedRadius };
-    }
 
-    findCurrentLocation();
+// Global variables for the map, markers, and info window.
+const markers: Map<string, google.maps.marker.AdvancedMarkerElement> =
+    new Map();
+let infoWindow: google.maps.InfoWindow;
+
+// The init function is called when the page loads.
+async function init(): Promise<void> {
+    // Import the necessary libraries from the Google Maps API.
+    const [{ InfoWindow }, { Place }, markerLib, coreLib] = await Promise.all([
+        google.maps.importLibrary('maps') as Promise<google.maps.MapsLibrary>,
+        google.maps.importLibrary(
+            'places'
+        ) as Promise<google.maps.PlacesLibrary>,
+        google.maps.importLibrary(
+            'marker'
+        ) as Promise<google.maps.MarkerLibrary>,
+        google.maps.importLibrary('core') as Promise<google.maps.CoreLibrary>,
+    ]);
+
+    const AdvancedMarkerElement = markerLib.AdvancedMarkerElement;
+
+    const LatLngBounds = coreLib.LatLngBounds;
+
+    // Create a new info window and set its content to the place details element.
+    infoWindow = new InfoWindow({
+        content: placeDetails,
+        ariaLabel: 'Place Details',
```

Comment:
```suggestion
        ariaLabel: 'Place Details',
        map: map.innerMap,
```

turns out this is needed otherwise a bunch of rendering glitches happen (e.g. when zooming, markers aren't locked to the map correctly)

Reply:
This actually threw a TS error because map is part of InfoWindowOpenOptions but not InfoWindowOptions